From 4ad3e08bde8ef787a71e587422f0b9a3b29e7c68 Mon Sep 17 00:00:00 2001
From: Zhihao Cheng <chengzhihao1@huawei.com>
Date: Mon, 23 May 2022 21:18:19 +0800
Subject: [PATCH] file: fix kabi change since add f_ctl_mode
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 4ad3e08bde8ef787a71e587422f0b9a3b29e7c68
Modified-by-SEL: Yes, modified due to different context


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I53R0H
CVE: NA
backport: openEuler-22.03-LTS

---------------------------

Signed-off-by: Zhihao Cheng <chengzhihao1@huawei.com>
Reviewed-by: Zhang Yi <yi.zhang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/fs.h | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/include/linux/fs.h b/include/linux/fs.h
index 44ee66086b94..18259e38dcd7 100644
--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -954,7 +954,6 @@ struct file {
 	atomic_long_t		f_count;
 	unsigned int 		f_flags;
 	fmode_t			f_mode;
-	fmode_t			f_ctl_mode;
 	struct mutex		f_pos_lock;
 	loff_t			f_pos;
 	struct fown_struct	f_owner;
@@ -976,8 +975,14 @@ struct file {
 	struct address_space	*f_mapping;
 	errseq_t		f_wb_err;
 	errseq_t		f_sb_err; /* for syncfs */
-
+#ifndef __GENKSYMS__
+	union {
+		fmode_t			f_ctl_mode;
+		u64			kabi_reserved1;
+	};
+#else
 	KABI_RESERVE(1)
+#endif
 } __randomize_layout
   __attribute__((aligned(4)));	/* lest something weird decides that 2 is OK */
 
-- 
2.34.1

